release: Zephyr v4.2.0

danieldegrasse opened the pull request:
Sets the version to v4.2.0, and removes v4.0.0 from the supported releases as it is now EOL

danieldegrasse committed:
doc: release: add v4.2.0 to the list of supported releases

Add 4.2.0 to the supported release, EOL to the tentative 4.4 release
date.

Signed-off-by: Dan Kalowsky <dkalowsky@amperecomputing.com>
Signed-off-by: Daniel DeGrasse <ddegrasse@tenstorrent.com>

danieldegrasse committed:
doc: release: Remove v4.0.0 from the list of supported releases

Only the two most recent mainline releases are supported.

Signed-off-by: Dan Kalowsky <dkalowsky@amperecomputing.com>
Signed-off-by: Daniel DeGrasse <ddegrasse@tenstorrent.com>

danieldegrasse committed:
doc: release: Finalize v4.2.0 release notes and migration guide

Remove working draft status from the v4.2.0 release notes and migration
guide.

Signed-off-by: Dan Kalowsky <dkalowsky@amperecomputing.com>
Signed-off-by: Daniel DeGrasse <ddegrasse@tenstorrent.com>

danieldegrasse committed:
release: Zephyr v4.2.0

Set the version to v4.2.0

Signed-off-by: Dan Kalowsky <dkalowsky@amperecomputing.com>
Signed-off-by: Daniel DeGrasse <ddegrasse@tenstorrent.com>

pdgendt commented:
Also update https://github.com/zephyrproject-rtos/zephyr/blob/main/.github/SECURITY.md

kartben commented:
> Also update [`main`/.github/SECURITY.md](https://github.com/zephyrproject-rtos/zephyr/blob/main/.github/SECURITY.md?rgh-link-date=2025-07-18T07%3A15%3A30Z)

Given it already has a typo (mentions 4.0 instead of 4.1 as "latest release" _and_ these PRs are typically used as templates from one release to the other, I would really really encourage to actually address this comment right in this PR

danieldegrasse committed:
release: update security policy to include 4.2

Update security policy documentation to indicate 4.2 is the latest
version, and drop 4.0 from the supported versions

Signed-off-by: Dan Kalowsky <dkalowsky@amperecomputing.com>
Signed-off-by: Daniel DeGrasse <ddegrasse@tenstorrent.com>
